tests/Google/Ads/GoogleAds/Lib/V6/GoogleAdsLoggingServerStreamingCallTest.php

```diff
@@ -27,39 +30,112 @@
 /**
  * Unit tests for `GoogleAdsLoggingServerStreamingCall`.
  *
+ * Note: With the limitation of PHPUnit, we cannot combine `@depends` and `@dataProvider` in a way
+ * that makes it work properly, so we need to implement two methods: `testGetStatus` and
+ * `testGetStatusWithEmptyResponses`, although both share the same code.
+ *
  * @covers \Google\Ads\GoogleAds\Lib\V6\GoogleAdsLoggingServerStreamingCall
  * @small
  */
 class GoogleAdsLoggingServerStreamingCallTest extends TestCase
 {
 
-    public function testGetStatus()
+    public function testResponses()
     {
         // Prepares the inner call.
-        $expectedStatus = new \stdClass();
-        $expectedStatus->code = 0;
-        $expectedStatus->message = 'success';
-        $expectedStatus->metadata = ['request-id' => ['AbCDeF']];
         $forwardingCallMock = $this->getMockBuilder(ForwardingServerStreamingCall::class)
             ->disableOriginalConstructor()
             ->getMock();
-        $forwardingCallMock->method('getStatus')->willReturn($expectedStatus);
+        $response1 = new UploadCallConversionsResponse();
+        $response2 = new UploadClickConversionsResponse();
+        $forwardingCallMock->method('responses')->willReturn([$response1, $response2]);
+
+        // Instantiates an GoogleAdsLoggingServerStreamingCall.
+        $googleAdsLoggingServerStreamingCall =
+            $this->createGoogleAdsLoggingServerStreamingCall($forwardingCallMock);
+
+        // Checks if the returned results are the same as those in the inner call;
+        $actualResponses = iterator_to_array($googleAdsLoggingServerStreamingCall->responses());
+        $this->assertEquals([$response1, $response2], $actualResponses);
+
+        return [
+            'forwardingCallMock' => $forwardingCallMock,
+            'streamingCall' => $googleAdsLoggingServerStreamingCall
+        ];
+    }
+
+    /**
+     * @depends testResponses
+     */
+    public function testGetStatus($dependingData)
+    {
+        $this->performStatusAssertion($dependingData);
+    }
+
+    public function testResponsesWithEmptyResponses()
```

Comment:
I think you should replace this test with additional test(s) in the `LogMessageFormatterTest` that are based on `ServerStreamingCall` mock (see what is currently done for [UnaryCall](https://github.com/googleads/google-ads-php/blob/master/tests/Google/Ads/GoogleAds/Lib/V6/LogMessageFormatterTest.php#L678)).

Reply:
I can add more test in the LogMessageFormatterTest, but this still needs to be here.
The reason is we need to test getStatus when the response is empty. Without this method to depend on, we'll need to copy most code from the testResponses() into getStatus(), which I want to avoid.

Reply:
SG. I was suggesting new tests because I believe the [empty/non-empty array logic](https://github.com/googleads/google-ads-php/pull/474/files#diff-d4a23076f7ae8234187fabb56741ae0dd14ca146afcd1f55fd2776a60dfbd9e7R57) is not entirely covered as of now.